src/bin/main.rs

```diff
@@ -157,39 +89,71 @@ fn exec_command(args: Vec<String>, setcap: bool) {
 }
 
 fn main() {
+    let attach_command = SubCommand::with_name("attach")
+        .about("Enter container")
+        .version(crate_version!())
+        .author(crate_authors!("\n"))
+        .setting(AppSettings::DisableVersion)
+        .arg(
+            Arg::with_name("effective-user")
+                .long("effective-user")
+                .takes_value(true)
+                .empty_values(false)
+                .value_name("EFFECTIVE_USER")
+                .help("effective username that should be owner of new created files on the host"),
+        )
+        .arg(
+            Arg::with_name("type")
+                .short("t")
+                .long("type")
+                .takes_value(true)
+                .empty_values(false)
+                .require_delimiter(true)
+                .value_name("TYPE")
+                .help("Container types to try (sperated by ','). [default: all but command]")
+                .possible_values(&ContainerType::variants()),
+        )
+        .arg(
+            Arg::with_name("id")
+                .help("container id, container name or process id")
+                .required(true)
+                .index(1),
+        )
+        .arg(command_arg(2));
+
+    let exec_command = SubCommand::with_name("exec")
+        .about("Execute command in container filesystem")
+        .version(crate_version!())
+        .author(crate_authors!("\n"))
+        .arg(command_arg(1));
+
+    let main_app = App::new("Cntr")
+        .about("Enter or executed in container")
+        .version(crate_version!())
+        .author(crate_authors!("\n"))
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .subcommand(attach_command)
+        .subcommand(exec_command.clone());
+
+    // find and run subcommand/app
     match std::env::current_exe() {
         Ok(exe) => {
             if exe == Path::new(cntr::SETCAP_EXE) {
-                exec_command(env::args().collect::<Vec<String>>(), true)
+                let matches = exec_command.get_matches();
+                exec(&matches, true);
+            } else {
+                let matches = main_app.get_matches();
+                match matches.subcommand() {
+                    ("exec", Some(exec_matches)) => exec(exec_matches, true),
```

Comment:
Almost correct:

```suggestion
                    ("exec", Some(exec_matches)) => exec(exec_matches, false),
```

fixed and merged in https://github.com/Mic92/cntr/pull/44